pageserver/src/tenant.rs

```diff
@@ -518,7 +519,7 @@ pub struct OffloadedTimeline {
     /// If we offload a timeline, we keep around the remote client
     /// for the duration of the process. If we find it through the
     /// manifest, we don't construct it up until it's needed (deletion).
-    pub remote_client: Option<Arc<RemoteTimelineClient>>,
+    pub remote_client: OnceLock<Arc<RemoteTimelineClient>>,
```

Comment:
This field is kind of confusing: sometimes it holds a `RemoteTimelineClient` that was copied from an initialized Timeline (i.e. has a live UploadQueue with layer metadata in it), but sometimes it contains an empty instance directly from build_timeline_client.

- I think the existing timeline deletion path might be buggy because of that -- I can't see a place where it properly initializes the RemoteTimelineClient if it constructs it (i.e. in the case that a timeline was offloaded since startup and has a None here).
- In the shard split case, we don't have any problem with constructing an empty one, because `download_index_file` doesn't use any UploadQueue state.

I would suggest in this PR that we avoid using OffloadedTimeline.remote_client at all, and just build a client directly in the shard split code, and then in a separate PR address the general issue (if I'm right) that remote_client can be in a state where it won't work for deletion.

Reply:
Yeah you are right, I just added a test case that deletes a timeline that has been offloaded and indeed, it complains about the upload queue being uninitialized.

It's been one of my todos in #8088 to add a test for offloaded deletion already, I suppose it also needs to be fixed.

Reply:
I have now removed the `OnceLock` changes and always construct a remote timeline client in the case of an offloaded timeline, not distinguishing between offloaded in the same process and offloaded in an earlier process.